resources/compass/templates/agent-configuration-job.yaml

```diff
@@ -0,0 +1,104 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: compass-agent-configuration
+  namespace: compass-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: compass-agent-configuration
+  namespace: compass-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: compass-agent-configuration
+subjects:
+  - kind: ServiceAccount
+    name: compass-agent-configuration
+    namespace: compass-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: compass-agent-configuration
+  namespace: compass-system
+rules:
+  - apiGroups: [""]
+    resources: [configmaps]
+    verbs: [create, list, get, watch]
```

Comment:
You don't need other permissions than creating configmaps, so I would remove other verbs